.github/workflows/check.yml

```diff
@@ -42,7 +42,7 @@ jobs:
         with:
           node-version: lts/jod
       - name: Set up JDK 17
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```

Comment:
Great catch. I'd appreciate if we could contain these drive-by fixes in their own (easy to merge) PRs 🙏 Reason being it's much easier to revert a PR like that if it doesn't also contain feature work.